Node.js update

Shanzid01 opened the pull request:
Updates for easy upload of json data to Cloud MongoDB.

Shanzid01 committed:
Added scripts to upload data

Shanzid01 committed:
All existing data upload

Tasmin153 committed:
Merge branch 'Node.js-update' of https://github.com/TheRSquaredProject/zilla-upazilla-db

Shanzid01 committed:
How to guide

Tasmin153 committed:
Merge branch 'Node.js-update' of https://github.com/TheRSquaredProject/zilla-upazilla-db

Tasmin153 committed:
no message

azmainamin commented on node.js_mongodb/how to use.txt:
```diff
@@ -0,0 +1,29 @@
+The update comes with two additions to upload data into MongoDB: SingleUpload.js & BulkUpload.js
```
This is great documentation! Thanks for adding this. Can you rename it to README.md?

azmainamin commented on node.js_mongodb/SingleUpload.js:
```diff
@@ -0,0 +1,24 @@
+PromptAndUpload();
+
+async function PromptAndUpload() {
+    const rl = readline.createInterface({
```
Let's use a more descriptive name than r1. I have no clue what this is supposed to be.

azmainamin commented on node.js_mongodb/SingleUpload.js:
```diff
@@ -0,0 +1,24 @@
+const DBHelper=require('./DB_helper');
+const readline = require('readline');
+PromptAndUpload();
+
```
Also, let's add some docstring to the functions i.e.
/*
* Adds a new collection if it does not exist. If collection exists, replaces it with the given collection
*
*/

azmainamin commented on node.js_mongodb/SingleUpload.js:
```diff
@@ -0,0 +1,24 @@
+            console.log("Overwriting data..");
+            let db="rsquared_zilla_upazilla_db";
+            var all_data = require('./data/'+jsonFileName);
+            let deleteExisting= await DBHelper.deleteData(db,collectionName,{});
```
Will it throw an error if collection does not exist? Can we make it more defensive so that it tries to delete only if it exists?

azmainamin commented on node.js_mongodb/app.js:
```diff
@@ -27,8 +28,19 @@ async function RunProgram(){
+    for(var i=0;i<jsonFileName.length; i++){
+        let db="rsquared_zilla_upazilla_db";
+        var all_data = require('./data/'+jsonFileName[i]+".json");
+        let deleteExisting= await DBHelper.deleteData(db,collectionName[i],{});
```
Same comment as checking to see if coll exists before trying to delete it.

azmainamin commented on node.js_mongodb/bulkupload.js:
```diff
@@ -0,0 +1,30 @@
+      rl.close();
+  });
+
+async function bulkUpload(){
```
Is this bulkUpload any different from the bulkUpload() func in app.js? Do we need both?

azmainamin commented on node.js_mongodb/bulkupload.js:
```diff
@@ -0,0 +1,30 @@
+    output: process.stdout
+  });
+  console.log("Attempting to upload all files in '/data' directory to mongoDB cloud.")
+  rl.question('Are you sure you want to upload and overwrite all data? (Y/N) ', async (reply) => {
```
Rename r1 to something more descriptive.

Shanzid01 committed:
Addressed all comments in PR

- Throws error and exists if file name does not exist in 'singleupload.js' and 'bulkupload.js'
- overwrites collection only if it exists, otherwise just creates new collection
- inserted comments
- changed input reader's variable name from 'rl' to 'input_reader'

Shanzid01 committed:
Removed old 'how to use.txt'

Shanzid01 committed:
Merge branch 'master' into Node.js-update
